File: Python/Programmers/약수의개수와덧셈.py
def solution(left, right):
    answer = 0
    for i in range(left, right+1):
        count = 0
        for j in range(1, i+1):
            if i % j == 0:
                count += 1

        if count % 2 == 0:
            answer += i
        else:
            answer -= i
    return answer

solution(13, 17)

File: Swift/PRG/약수의개수와덧셈.swift
//
//  main.swift
//  약수의 개수와 덧셈 - level1
//
//  Created by heerucan on 2022/08/14.
//

import Foundation


func solution(_ left:Int, _ right:Int) -> Int {
    func findCount(_ num: Int) -> Int {
        return Array(1...num).filter{num % $0 == 0}.count
    }
    
    var sum = 0

    for i in left...right {
        if findCount(i) % 2 == 0 {
            sum += i
        } else {
            sum -= i
        }
    }
    return sum
}

solution(13, 17)
